PlayList.java

```diff
@@ -70,48 +91,100 @@ public int indexOf(String title) {
      *  If i is negative or greater than the size of this list, or if the list
      *  is full, does nothing and returns false. Otherwise, inserts the track and
      *  returns true. */
-    public boolean add(int i, Track track) {
-        //// replace the following statement with your code
-        return false;
+    public boolean add(int i, Track track) 
+    {
+        if(this.size>=this.maxSize||i>=maxSize)
+            return false;
+        for (int j=this.size;j>i;j-- ) 
+        {
+            this.tracks[j]=this.tracks[j-1];
+        }
+        this.tracks[i]=track;
+        this.size++;
+        return true;
+
     }
      
     /** Removes the track in the given index from this list.
      *  If the list is empty, or the given index is negative or too big for this list, 
      *  does nothing and returns -1. */
-    public void remove(int i) {
-        //// replace this comment with your code
+    public void remove(int i) 
+    {
+        if(this.size==0||i>=this.maxSize)
+            return;
+        this.tracks[i]=null; 
+        for (int k=i+1;k<size;k++ ) 
+        {
+            this.tracks[k-1]=this.tracks[k];
+        }
+        this.tracks[size-1]=null;
+        this.size--;
+
     }
 
     /** Removes the first track that has the given title from this list.
      *  If such a track is not found, or the list is empty, or the given index
      *  is negative or too big for this list, does nothing. */
-    public void remove(String title) {
-        //// replace this comment with your code
+    public void remove(String title) 
+    {
+        int index = this.indexOf(title);
+        if(index>-1)
+            this.remove(index);
     }
 
     /** Removes the first track from this list. If the list is empty, does nothing. */
-    public void removeFirst() {
-        //// replace this comment with your code
+    public void removeFirst() 
+    {
+        if(this.tracks[0]!=null)
+        {
+            this.tracks[0]=null;
+            this.size--;  
+        }
+        for (int i=1;i<size+1 ;i++ ) 
+        {
+            this.tracks[i-1]=this.tracks[i];
+        }
     }
     
     /** Adds all the tracks in the other list to the end of this list. 
      *  If the total size of both lists is too large, does nothing. */
     //// An elegant and terribly inefficient implementation.
-     public void add(PlayList other) {
-        //// replace this comment with your code
-    }
+     public void add(PlayList other) 
+     {
+        if(this.maxSize<=this.size)
```

Comment:
The method should validate that it can contain the whole given playlist via param.
```suggestion
        if(this.maxSize<this.size + other.size)
```